workflow/rules/train_model.smk

```diff
@@ -105,6 +125,40 @@ rule train_model:
 			--feature_table_file {input.feature_table} \
 			--out_dir {params.out_dir} \
 			--polynomial {params.polynomial} \
-			--epsilon {params.epsilon}
+			--epsilon {params.epsilon} \
+			--params_file {input.model_params}
 		"""
 		
+# compare cv-performance on training data across all models (note, this is not the true benchmarking performance CRISPR elements not overlapping prediction elements aren't considered)  
+rule gather_model_performances:
```

Comment:
For the future, you can consider splitting the actual model training components from the best model selection components into separate snakemake modules to make things cleaner.